wagtail/admin/localization.py

```diff
@@ -117,3 +120,22 @@ def get_available_admin_time_zones():
     return getattr(
         settings, "WAGTAIL_USER_TIME_ZONES", sorted(zoneinfo.available_timezones())
     )
+
+
+def gettext_domain(domain: str, message: str) -> str:
+    """
+    Similar to Django's ``gettext``, return the translation for a specific
+    domain (i.e. a specific ``.po`` file).
+    """
+    if settings.USE_I18N:
+        # Get current language, and fallback to non-specific version of language.
+        langs = [get_language()] + [get_language().split("-")[0]]
+        lc_path = os.path.join(os.path.dirname(__file__), "locale")
+        _gt = gettext.translation(
+            domain,
+            localedir=lc_path,
+            languages=langs,
+            fallback=True,
+        )
+        return _gt.gettext(message)
```

Comment:
I have a slight concern that this isn't very performant, since we reconstruct the translation object every time we call `gettext_domain()`. [Django's implementation](https://github.com/django/django/blob/main/django/utils/translation/trans_real.py) has quite a lot of work put into caching the translations per thread, so I assume it could get a bit heavy. Especially since we call this function for every timezone when loading the accounts view.

I wonder whether we can leverage Django's implementation, or maybe somehow make the strings generated as part of the `django` domain generated for Wagtail by `makemessages`. Failing that, maybe add a simple cache that keeps the translation object for each locale.

Reply:
It's been too many years so now I forget why I implemented it this way 😵 But that is a valid point - surely there must be a way to hook into Django's translations.